SimG4CMS/Calo/src/CaloTrkProcessingBuilder.cc

```diff
@@ -0,0 +1,90 @@
+#include "FWCore/PluginManager/interface/ModuleDef.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+class CaloTrkProcessingBuilder : public SensitiveDetectorMakerBase {
+public:
+  explicit CaloTrkProcessingBuilder(edm::ParameterSet const& p, edm::ConsumesCollector cc)
+      : cspsToken_{cc.esConsumes<edm::Transition::BeginRun>()}, caloSimPar_{nullptr} {
+    bool dd4hep = p.getParameter<bool>("g4GeometryDD4hepSource");
+    addlevel_ = dd4hep ? 1 : 0;
+    edm::ParameterSet csps = p.getParameter<edm::ParameterSet>("CaloTrkProcessing");
+    testBeam_ = csps.getParameter<bool>("TestBeam");
+    eMin_ = csps.getParameter<double>("EminTrack") * CLHEP::MeV;
+    putHistory_ = csps.getParameter<bool>("PutHistory");
+    doFineCalo_ = csps.getParameter<bool>("DoFineCalo");
+    eMinFine_ = csps.getParameter<double>("EminFineTrack") * CLHEP::MeV;
+    fineNames_ = csps.getParameter<std::vector<std::string> >("FineCaloNames");
+    fineLevels_ = csps.getParameter<std::vector<int> >("FineCaloLevels");
+    useFines_ = csps.getParameter<std::vector<int> >("UseFineCalo");
+    for (auto& level : fineLevels_)
+      level += addlevel_;
+  }
+
+  void beginRun(const edm::EventSetup& es) final { caloSimPar_ = &es.getData(cspsToken_); }
+
+  std::unique_ptr<SensitiveDetector> make(const std::string& iname,
+                                          const SensitiveDetectorCatalog& clg,
+                                          const edm::ParameterSet& p,
+                                          const SimTrackManager* man,
+                                          SimActivityRegistry& reg) const final {
+    auto sd = std::make_unique<CaloTrkProcessing>(iname,
+                                                  *caloSimPar_,
+                                                  clg,
+                                                  testBeam_,
+                                                  eMin_,
+                                                  putHistory_,
+                                                  doFineCalo_,
+                                                  eMinFine_,
+                                                  addlevel_,
+                                                  fineNames_,
+                                                  fineLevels_,
+                                                  useFines_,
+                                                  man);
+    SimActivityRegistryEnroller::enroll(reg, sd.get());
+    return sd;
+  }
+
+private:
+  const edm::ESGetToken<CaloSimulationParameters, HcalParametersRcd> cspsToken_;
+  const CaloSimulationParameters* caloSimPar_;
+  bool testBeam_;
+  double eMin_;
+  bool putHistory_;
+  bool doFineCalo_;
+  double eMinFine_;
+  int addlevel_;
+  std::vector<std::string> fineNames_;
+  std::vector<int> fineLevels_;
+  std::vector<int> useFines_;
+};
+
+DEFINE_SENSITIVEDETECTORBUILDER(CaloTrkProcessingBuilder, CaloTrkProcessing);
```

Comment:
This file (or at minimum this line) should be in `plugins` instead of `src` (as any other plugin definition). This is causing following job failures in CLANG IBs
```
----- Begin Fatal Exception 16-Jul-2021 01:20:57 CEST-----------------------
An exception of category 'WrongPluginLoaded' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=OscarMTProducer label='g4SimHits'
Exception Message:
The plugin 'CaloTrkProcessing' should have been loaded from
 '/cvmfs/cms-ib.cern.ch/nweek-02689/slc7_amd64_gcc900/cms/cmssw-patch/CMSSW_12_0_CLANG_X_2021-07-15-2300/lib/slc7_amd64_gcc900/pluginSimG4CMSCaloPlugins.so'
 but instead it was already loaded from
 '/cvmfs/cms-ib.cern.ch/nweek-02689/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_0_CLANG_X_2021-07-14-2300/lib/slc7_amd64_gcc900/pluginSimG4CMSTestBeamPlugins.so'
 because some other plugin was loaded from the latter loadables.
To work around the problem the plugin 'CaloTrkProcessing' should only be defined in one of these loadables.
----- End Fatal Exception -------------------------------------------------
```